frontend/src/components/QueryBuilder/QueryBuilder.tsx

```diff
@@ -1427,315 +1353,60 @@ function QueryBuilder<T extends SearchParamsInput | ErrorSearchParamsInput>(
-												lte:
-													Number(
-														value?.split('_')[1],
-													) *
-													60 *
-													1000,
-										  }),
-								},
-							},
-						}
-					case 'between':
-						return {
-							range: {
-								[name]: {
-									gte: Number(value?.split('_')[0]),
-									...(Number(value?.split('_')[1]) ===
-									RANGE_MAX_LENGTH
-										? null
-										: {
-												lte: Number(
-													value?.split('_')[1],
-												),
-										  }),
-								},
-							},
-						}
-				}
-			} else {
-				const key = field.value
-				switch (op) {
-					case 'is':
-						return {
-							term: { 'fields.KeyValue': `${key}_${value}` },
-						}
-					case 'contains':
-						return {
-							wildcard: {
-								'fields.KeyValue': `${key}_*${value}*`,
-							},
-						}
-					case 'matches':
-						return {
-							regexp: {
-								'fields.KeyValue': `${key}_${value}`,
-							},
-						}
-					case 'exists':
-						return { term: { 'fields.Key': key } }
-				}
-			}
-		},
-		[getCustomFieldOptions, admin],
-	)
-
-	const parseRuleImpl = useCallback(
-		(
-			field: SelectOption,
-			op: Operator,
-			multiValue: MultiselectOption,
-		): any => {
-			if (isNegative(op)) {
-				return {
-					bool: {
-						must_not: {
-							...parseRuleImpl(
-								field,
-								NEGATION_MAP[op],
-								multiValue,
-							),
-						},
-					},
-				}
-			} else if (hasArguments(op)) {
-				return {
-					bool: {
-						should: multiValue.options.map(({ value }) =>
-							parseInner(field, op, value),
-						),
-					},
-				}
-			} else {
-				return parseInner(field, op)
-			}
-		},
-		[parseInner],
-	)
-
-	const parseRule = useCallback(
-		(rule: RuleProps): any => {
-			const field = rule.field!
-			const multiValue = rule.val!
-			const op = rule.op!
-
-			return parseRuleImpl(field, op, multiValue)
-		},
-		[parseRuleImpl],
-	)
 	const { data: appVersionData } = useGetAppVersionsQuery({
 		variables: { project_id: projectId! },
 		skip: !projectId,
 	})
 
 	const [currentRule, setCurrentRule] = useState<RuleProps | undefined>()
+
+	const {
+		isAnd: serializedIsAnd,
+		rules: serializedRules,
+	}: { isAnd: boolean; rules: any } = JSON.parse(searchQuery)
+	const startingRules = deserializeRules(serializedRules)
+	const [isAnd, toggleIsAnd] = useToggle(serializedIsAnd)
+	const [rules, setRules] = useState<RuleProps[]>(startingRules)
+
+	const startingDateRange = startingRules.find(
+		(rule) =>
+			rule.op === 'between_date' &&
+			rule.field?.value === timeRangeField.value,
+	)?.val?.options?.[0]?.value
+	let from, to: Date | undefined
+	if (startingDateRange) {
+		const [fromStr, toStr] = startingDateRange.split('_')
+		from = new Date(fromStr)
+		to = new Date(toStr)
+	}
 	const [dateRange, setDateRange] = useState<Date[]>([
-		defaultPreset.startDate, // Start at 30days
-		new Date(now.toISOString()),
+		from ?? defaultPreset.startDate, // Start at 30days
+		to ?? new Date(getNow().toISOString()),
```

Comment:
`getNow` is a little hacky, but we were using a slightly different `now` date in a few places which broke logic comparing `now` with the relative dates that required them to be exactly the same